### DISCUSSION TITLE
When an issue/PR is referred to in a discussion, mention that in that referenced issue/PR

### DISCUSSION THREAD

**Discussion by pubudu91** [community/community#3193] [Discussions]
When an issue/PR is referred in another issue/PR, it shows up in that issue/PR saying this issue/PR was referenced in this particular issue/PR. Currently this doesn't seem to happen when we refer to issues (didn't check, but I'm guessing it'd be the same with PRs) in discussions. It would be great if mentions in discussions are also included so that someone looking at the issue can see the discussions where it was referenced.

**Comment by jay** [comment]
Totally agree, this is a very useful feature. I also submitted a github request a while ago asking for locked issues to allow references. Often we want to stop the comments but allow references to continue since they help during any investigation where an old issue needs to be revisited.

**Reply by jay** [reply]
This just bit me again today. We have a discussion [here](https://github.com/curl/curl/discussions/8585) and I spent time researching the report only to discover the reporter filed an issue with OpenSSL [here](https://github.com/openssl/openssl/issues/17875). Had I known that (ie had there been reference in the discussion, since the ossl issue links the discussion) it would have saved me some time.

**Comment by brandonkelly** [comment]
Related, discussions should also show their own mentions, from other issues/discussions. Just posted a new thread for that: #4301
